FrameView: add a more functional approach to iterate rows/columns (exploration)

ryantxu opened the pull request:
This is an experiment to see if we can have a more functional appraoch... feedback welcome.

This lets you iterate frame row/field values with:

```
const vals = view.rows.map((fields, index) => {
      return fields.map(f => {
        return f.value;
      });
    });
```

or

```
for (const fields of view.rows) {
      vals2.push(
        fields.map(f => {
          return f.value;
        })
      );
    }
```

marcusolsson commented:
Sorry for the delay. I'll play around with this first thing tomorrow! 👍

ryantxu commented:
@marcusolsson -- something we should consider for 7.3?

marcusolsson commented on packages/grafana-data/src/dataframe/DataFrameView.ts:
```diff
@@ -1,7 +1,12 @@
 
+interface FieldAndValue {
+  field: Field;
+  value: any;
```
This is handy, but makes it a bit weird to use:

```ts
view.rows.map((fields, index) => {
  return fields.map(field => field.field); // field.field?
});
```

What if `fields` was a `Field[]`, and you'd still use the index? Might not be reason enough to add it.

```ts
view.rows.map((fields, index) => {
  return fields.map(field => field.values.get(index));
});
```

Or—forgive my TypeScript skills here—could we return a union type that just adds add `value` property to the Field? Might be too much magic though.

anthonyeleven committed:
docs: a few tweaks for clarity and readability (#28579)

Signed-off-by: Anthony D'Atri <anthony.datri@gmail.com>

hugohaggmark committed:
StatPanel: Fixes BizChart error max: yyy should not be less than min zzz (#28587)

ryantxu committed:
AddDatasource: Improve plugin categories (#28584)

* add IoT category to datasources

* add more enterprise plugins

* add more enterprise plugins

Clarity-89 committed:
Grafana-UI: Add Card component (#28216)

* Base card

* Add disabled state

* Expand knobs

* Add card actions

* Add meta data

* Allow custom tags

* Extend container props

* Add inner link

* Add docs

* Add missing keys

* Update margins

* Add description

* Add full card example

* Tweak disabld state

* Export Card

* title => heading

* Filter out empty content

* Add disableEvents

* Move tooltip to container

* Use new Card for AlertRuleItem

* Cleanup

* Update snapshot

* Rename props

* Rename props[2]

* Disable hover is onClick is missing

* Fix alert rule item

* Update snapshot

* Export CardProps

* Replace logo

* Remove tag prop

* Remove extra div

* Add @public

* Update AlertRuleItem

* Simplify disabled logic

* Export Card styles

* Remove AlertRuleItem tooltips

* Revert to old button design

* Make component internal

Co-authored-by: Torkel Ödegaard <torkel@grafana.com>

ivanahuckova committed:
Explore/Loki: Update docs and cheatsheet (#28541)

* Add updated histogram docs

* Add to cheatsheet

* Update

* Update docs/sources/explore/index.md

* Update docs/sources/explore/index.md

kaydelaney committed:
CloudWatch Logs queue and websocket support (#28176)

CloudWatch Logs queue and websocket support

aknuds1 committed:
CI: Add GCC to Windows Docker image (#28562)

* CI: Add GCC to Windows Docker image

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* CI: Upgrade golangci-lint in Windows Docker image

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

marefr committed:
Docs: Additional 7.3 upgrade notes (#28592)

Adds a couple upgrade notes for v7.3 regarding database migrations for 
user invites and snapshots.

Co-authored-by: Will Browne <wbrowne@users.noreply.github.com>

marefr committed:
Docs: Data source provisioning and sigV4 (#28593)

Document the sigv4 properties for data source provisioning.

clyang82 committed:
Dashboard: Allow add panel for viewers_can_edit (#28570)

Signed-off-by: clyang82 <chuyang@redhat.com>

mckn committed:
Live: updated the reference to use lazy loaded Monaco in code editor. (#28597)

dprokop committed:
Update uPlot to 1.2.2 and align timestamps config with new uPLot API (#28569)

kylebrandt committed:
changelog: update for 7.3.0 (#28602)

isaozlerfm committed:
Field config API: add slider editor (#28007)

* Field config: implementation slider editor (#27592)

* PR-28007 feedback fixed

* Field config: implementation slider editor (#27592)

* PR-28007 feedback fixed

* processed review PR-28007

* Field config: implementation slider editor (#27592)

* PR-28007 feedback fixed

* Field config: implementation slider editor (#27592)

* processed review PR-28007

* fixing leftover number[] bugs

* RichHistoryQueriesTab.test fix + slider vertical feat fixed

* fixed Slider.test.tsx expectation

* Added @docs to prevent build-frontend-docs from failing

Co-authored-by: Isa Ozler <contactme@isaozler.com>

xlson committed:
Docs: data source insights (#28542)

* Docs: data source insights

* wording

* review feedback

* screenshot

* feedback

kylebrandt committed:
update latest.json (#28603)

oddlittlebird committed:
Docs: Update graph panel for tabs (#28552)

* Update data-links.md

* Update data-links.md

* content updates

* Update data-links.md

aknuds1 committed:
Developer guide: Update wrt. Windows (#28559)

* Developer guide: Update wrt. Windows

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update contribute/developer-guide.md

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Update contribute/developer-guide.md

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

aknuds1 committed:
CloudWatch: Improve method name, performance optimization (#28632)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

hborchardt committed:
Dashboard: Fix navigation from one SoloPanelPage to another one (#28578)

* Add test for SoloPanelPage.tsx

* Fix navigation from one SoloPanelPage to another one

The panel did not update because it assumed that the dashboard was already
fully loaded.

dprokop committed:
Plugin signing: Fix copy on signed plugin notice (#28633)

* Fix copy on signed plugin notice

* Update public/app/features/plugins/PluginPage.tsx

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update public/app/features/plugins/PluginPage.tsx

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

xlson committed:
Build: support custom build tags (#28609)

* Build: support custom build tags

* Update build.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

torkelo committed:
Table: Fix image cell mode so that it works with value mappings (#28644)

torkelo committed:
StatPanel: Fixed value being under graph and reduced likley hood for white and dark value text mixing (#28641)

* StatPanel: Fixed value being under graph and reduced likley hood for white and dark value text mixing

* Updated snapshot

* Updated storybook config

dprokop committed:
Bump rxjs to 6.6.3 (#28657)

chugunov committed:
Docker:  use root group in the custom Dockerfile (#28639)

wbrowne committed:
Plugins: Fix descendent frontend plugin signature validation (#28638)

* move plugin root check to earlier in validation process

* remove comment

* only check root if necessary

Commit:
Docs: Describe pipeline aggregation changes in v7.3 (#28660)

dprokop committed:
Add info about CSV download for Excel in What's new article (#28661)

* Add info about CSV download for Excel in What's new article

* Update docs/sources/whatsnew/whats-new-in-v7-3.md

* Update docs/sources/whatsnew/whats-new-in-v7-3.md

zoltanbedi committed:
Cloudwatch: Fix duplicate metric data (#28642)

* Cloudwatch: Fix duplicate metric data

* Refactor reduce function to for of

torkelo committed:
PanelMenu: Fixes panel submenu not being accessible for panels close to the right edge of the screen (#28666)

* Dropdowns: Trying to fix dropdown menus

* Dropdowns: Trying to fix dropdown menus

* removed now unnessary wrapper ref

* Upodates

* Remove export

torkelo committed:
TextPanel: Fixes problems where text panel would show old content (#28643)

zoltanbedi committed:
Chore: allow higher nodejs version than 12 (#28624)

ivanahuckova committed:
Loki: Fix error when some queries return zero results (#28645)

* Fix logSeriesToLogsModel when some data frames have empty fields

* Update test

* Update public/app/core/logs_model.ts

Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

aocenas committed:
Prometheus: Fix copy paste behaving as cut and paste (#28622)

mckn committed:
Build: changing docs docker image to prevent setting up frontend devenv. (#28670)

* Build: changing docs dev-environment

I have added a new image that will include the auto-generated frontend docs so you don't have to generate those locally prior to running the doc container. This has a dependency on this PR: https://github.com/grafana/website/pull/2744 and that we publish a version of the `grafana/docs-devenv` image.

* will ignore everything (even symbolic links)

* fixed spelling.

* added newline.

* updated image name.

aknuds1 committed:
Drone: Don't build on Windows for PRs (#28663)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

dprokop committed:
Changelog update - 7.3.1 (#28699)

* Changelog update

* Update CHANGELOG.md

* Update CHANGELOG.md

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

dprokop committed:
Update latest version to 7.3.1 (#28701)

aocenas committed:
Chore: Add some scenario tests for Explore (#28534)

* Add tests for main scenarios

* Add tests for split

* Remove unnecessary code

* Fix opening datasource picker

* Mock missing function

achatterjee-grafana committed:
updates from https://github.com/grafana/grafana/pull/28679 (#28708)

ceh committed:
Chore: Convert sqlstore annotation test from GoConvey to testify (#28715)

* Chore: Convert sqlstore annotation test from GoConvey to testify

* Use require.NoError

* Use assert.Empty

* Use assert.NoError in test cleanup

eamonryan committed:
Fix typo in unsigned plugin warning (#28709)

Typo fix!

hugohaggmark committed:
Templating: Speeds up certain variable queries for Postgres, MySql and MSSql (#28686)

hugohaggmark committed:
TableFilters: Fixes filtering with field overrides (#28690)

* TableFilters: Fixes filtering with field overrides

* Refactor: changes after PR comments

aknuds1 committed:
Drone: Upgrade build-pipeline tool (#28728)

* Drone: Upgrade build-pipeline tool

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Drone: Raise Drone's memory limit for build-storybook

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Chore: Pin nginx base image in nginx proxy Dockerfiles (#28730)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

Clarity-89 committed:
AlertRuleList: Add keys to alert rule items (#28735)

torkelo committed:
Dependencies: Remove unused dependency (#28711)

* Dependencies: Remove unused dependency

* Remove import

* Updated

jackw committed:
Migration: Remove LegacyForms from dashboard folder permissions (#28564)

* refactor(permissionlist): replace legacyform components with next-gen equivalents

* refactor(permissionlist): replace legacyform components

* refactor(permissionlist): clean up PermissionsInfo

* feat(grafana-ui): define line-height for Select option descriptions

* refactor(permissionlist): remove debug code

* refactor(permissionlist): simplify component tree with horizontalgroup. Add label

* refactor(permissionlist): remove redundant classname from Select

dprokop committed:
Plugin page: Fix dom validation warning (#28737)

tabarra committed:
LoginPage: Removed auto-capitalization from the login form (#28716)

* Removed auto-capitalization from the login form

On mobile browsers, by default most keyboards will auto capitalize the first letter of the email/username, which is case sensitive.  
This PR fixes that.

* Fixed autoCapitalize react keyword

torkelo committed:
Provisioning: Fixed problem with getting started panel being added to custom home dashboard (#28750)

* Provisioning: Fixed problem with getting started panel being added to custom home dashboard

* Fixed comment

* Update pkg/api/dashboard.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

jackw committed:
Dashboard: fix view panel mode for Safari / iOS (#28702)

* fix(dashboard): flex-basis so child height percentages are respected

* fix(dashboard): reset scrollTop when entering view panel mode

ivanahuckova committed:
Explore: Remove redundant decodeURI and fix urls (#28697)

* Remove redundant path URIdecoding

* Remove redundant comments

aknuds1 committed:
gtime: Add ParseDuration function (#28525)

* gtime: Make calculations independent of current time

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Introduce gtime.ParseDuration function

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* gtime: Fix ParseDuration

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

achatterjee-grafana committed:
Reoeragnization. (#28760)

peterholmberg committed:
Docs: Add docs for InfoBox component (#28705)

* add mdx

* fixes after pr feedback

aknuds1 committed:
devenv: Upgrade MSSQL Docker image (#28749)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Drone: Upgrade build pipeline tool (#28769)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

dprokop committed:
grafana/toolkit: Extract CHANGELOG when building plugin (#28773)

torkelo committed:
Cloudwatch: Fix issue with field calculation transform not working properly with Cloudwatch data (#28761)

* Cloudwatch: Fix issue with reducer transform not working properly with Cloudwatch data

* Updated go sdk and updated code

* CloudWatch: Improve test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

ndeevy committed:
Docs: Made usage of config/configuration consistent #19270 (#28167)

Rebased ndeevy:docs-config/configuration-consistency-update from grafana/master to resolve conflicts

Co-authored-by: ndeevy <ndeevy@redhat.com>

zen committed:
Docs: Add NGINX example for using websockets to Loki (#27998)

* feat: added NGINX example for using websockets to Loki

* chore: corrections after review

* chore: more correrctions from review

* chore: creview corrections

* chore: addressed review comment by @achatterjee-grafana

peterholmberg committed:
ColorSchemes: Add new color scheme (#28719)

torkelo committed:
EventBus: Introduces new event bus with emitter backward compatible interface (#27564)

* updated

* Experimenting with event bus with legacy support

* Before switch to emitter

* EventBus & Emitter unification

* Everything using new EventBus

* Making progress

* Fixing merge issues

* Final merge issues

* Updated

* Updates

* Fix

* Updated

* Update

* Update

* Rename methods to publish and subscribe

* Ts fixes

* Updated

* updated

* fixing doc warnigns

* removed unused file

torkelo committed:
Dependencies: Update angularjs to 1.8.2 (#28736)

* Angular: Update to angukar v1.8 with backward compatability monkey patches

* Update fix

* Updated

* Updated

* Fixes

* Update wording

* Update

aknuds1 committed:
Plugin Dockerfiles: Upgrade Go, golangci-lint, gcloud SDK (#28767)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

mckn committed:
Docs: added code comments to frontend packages. (#28784)

* fixed docs linting errors.

* decreased the limit.

torkelo committed:
Events: Remove unused or unnecessary events  (#28783)

* Events: Removing unused or unnessary events

* More cleanup

* Updated

ryantxu committed:
Live: improve broadcast semantics and avoid double posting (#28765)

dprokop committed:
Adds storybook integrity check to drone config (#28785)

* Adds storybook integrity check to dron config

* Update starlark files

* Review comments

* Bump grabpl version

* Drone: Fix dangling dependency

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

torkelo committed:
Graph: Fixes event emit function error (#28795)

hugohaggmark committed:
Variables: Fixes URL values for dependent variables (#28798)

dprokop committed:
grafana/toolkit: Drop console and debugger statements by default when building plugin with toolkit (#28776)

dprokop committed:
Panel inspect: Interpolate variables in panel inspect title (#28779)

* Interpolate variables in panel inspect title

* Update public/app/features/dashboard/components/Inspector/InspectContent.tsx

Co-authored-by: Torkel Ödegaard <torkel@grafana.org>

* fix typo

Co-authored-by: Torkel Ödegaard <torkel@grafana.org>

yenalsnavaj committed:
Variables: Adds named capture groups to variable regex (#28625)

* Dashboard: Add named capture groups to variable query regex

Variable query regex are able to use 'text' and 'value' named capture
groups to allow for separate display text to be extracted from the
query result. e.g.

Using a regex of /foo="(?<text>[^"]+)|bar="(?<value>[^"]+)/g on a query
result of metric{foo="FOO", bar="BAR"} would result in the variable
value being set to 'BAR' but display text being set to 'FOO'

Resolves #21076

* Improve regex capture group documentation

* Update docs/sources/variables/filter-variables-with-regex.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Hugo Häggmark <hugo.haggmark@gmail.com>

* Use text capture if value capture does not match

This is to keep the behaviour consistent with the current behavior. See
discussion https://github.com/grafana/grafana/pull/28625/files#r516490942

* Improve regex field placeholder and tooltip message

To make the feature more discoverable to users the place holder example
now includes the named capture groups. The tool tip message also
includes a reference and link to the documentation.

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>
Co-authored-by: Hugo Häggmark <hugo.haggmark@gmail.com>

triptu committed:
Dashboard: Add Datetime local (No date if today) option in panel axes' units (#28011)

* Add Datetime local (No date if today) option

* Add more tests

* Revert to using function for local dateTime format
This is required as Intl api used to convert date to local format is not present in
node environments

jedstar committed:
StatPanels: Add new calculation option for percentage difference  (#26369)

* Update fieldReducer.ts

addition of percentage difference to the singlestat panel

* Update time_series2.ts

* Update module.ts

* Update calculations-list.md

* Update docs/sources/panels/calculations-list.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Update public/app/plugins/panel/singlestat/module.ts

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Update module.ts

* Update calculations-list.md

* Update calculations-list.md

* Update module.ts

* Update fieldReducer.ts

* Update fieldReducer.ts

* Update fieldReducer.test.ts

* change name to remove wildcard characters

* Update calculations-list.md

* Update time_series2.ts

Fix spelling

* Update module.ts

* Update fieldReducer.ts

* formatting

* Update fieldReducer.ts

* Update fieldReducer.test.ts

* Update fieldReducer.test.ts

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

dprokop committed:
Remove homelinks panel (#28808)

ivanahuckova committed:
Icon: Replace font awesome icons where possible (#28757)

* Replace font awesome icons where possible

* Implement small updates

dprokop committed:
core and grafana/toolkit: Use latest version of grafana-eslint-conifg (#28816)

dprokop committed:
grafana/toolkit: allow builds with lint warnings (#28810)

* grafana/toolkit: allow builds with lint warnings

* Remove redundant code

torkelo committed:
Gauge: Improve font size auto sizing (#28797)

* Gauge: Improved font size calculations

* Added some comments

* update

* Moving to variable

wbrowne committed:
increase blob column size for encrypted dashboard data (#28831)

torkelo committed:
Alert: Remove z-index on Alert component so that it does not overlay ontop of other content (#28834)

dslizardo committed:
Currency: add Philippine peso currency (PHP) (#28823)

torkelo committed:
Marked: Upgrade and always sanitize by default (#28796)

* Marked: Upgrade and always sanitize by default

* Added test

* corrected text panel logic

hugohaggmark committed:
Variables: Adds variables inspection (#25214)

* Chore: initial commit

* Refactor: adds usages

* Refactor: adds FeatureInfoBox

* Refactor: uses graphs instead

* Refactor: adds id, name, title to prop instead of key

* Refactor: adds usages to VariablesList

* Refactor: Moves unknowns to first page

* Refactor: minor stylings and icons

* Refactor: styling

* Refactor: changed to button with modal

* Refactor: adds VariablesDependenciesButton

* Refactor: changes after UX feedback

* Refactor: renames heading

* Refactor: changes after PR comments

* Refactor: small changes after PR comments

mckn committed:
Units: added support to handle negative fractional numbers. (#28849)

aknuds1 committed:
Chore: Remove dead code (#28664)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Chore: Fix issues found by staticcheck (#28802)

* Fix linting issues

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

zoltanbedi committed:
Prometheus: fix missing labels from value (#28842)

zoltanbedi committed:
Tempo: fix test data source (#28836)

Clarity-89 committed:
Disable selecting enterprise plugins with no license (#28758)

* Add unlicensed property to plugins

* Disable selecting unlicensed plugin

* Add customizable plugin market place url

* License: workaround enabled only in enterprise

* linter

* Move licensing info to front end

* Update pkg/services/licensing/oss.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/services/licensing/oss.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/setting/setting.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/setting/setting.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/api/frontendsettings.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update sample.ini

* Update docs

* Update packages/grafana-runtime/src/config.ts

Co-authored-by: Torkel Ödegaard <torkel@grafana.org>

* Update public/app/features/datasources/state/buildCategories.ts

Co-authored-by: Torkel Ödegaard <torkel@grafana.org>

* Update pkg/api/frontendsettings.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/setting/setting.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix spelling

Co-authored-by: Leonard Gram <leo@xlson.com>
Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>
Co-authored-by: Torkel Ödegaard <torkel@grafana.org>

aknuds1 committed:
Chore: Fix staticcheck issues (#28854)

* Chore: Fix issues reported by staticcheck

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Undo changes

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Chore: Fix staticcheck issues (#28860)

* Chore: Fix issues reported by staticcheck

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Undo changes

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Chore: Fix issues reported by staticcheck

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

torkelo committed:
VizLayout: Simple viz layout component for legend placement and scaling (#28820)

* Updates

* Updated

* Rename slot to legend

* Updated

* Adding js doc comments

* hide horizontal track

* Fix docs error

* Added another export

* removing auto mode

* I hate these doc warnings

Rowayda-Khayri committed:
Fix titles case and add missing punctuation marks (#28713)

* Fix titles case and add missing punctuation marks

* Revert title case

mckn committed:
Variables: added __user.email to global variable (#28853)

* added email to global variable.

* forgot to increase the count.

simianhacker committed:
Elasticsearch: Filter pipeline aggregations from order by options (#28620)

aknuds1 committed:
Chore: Fix issues reported by staticcheck; enable stylecheck linter (#28866)

* Chore: Fix issues reported by staticcheck

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Apply suggestions from code review

Co-authored-by: Emil Tullstedt <emil.tullstedt@grafana.com>

marcusolsson committed:
Docs: Plugin signing docs (#28671)

* WIP

* Update plugin signing docs

* Fix review comments

ryantxu committed:
PanelHeader: show streaming indicator (and allow unsubscribe)  (#28682)

aknuds1 committed:
CRLF (#28822)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

achatterjee-grafana committed:
Update share-panel.md (#28880)

ryantxu committed:
Live: update centrifuge and the ChannelHandler api (#28843)

mjseaman committed:
Docs: Update list of Enterprise plugins (#28882)

systemcrash committed:
Loki: Correct grammar in DerivedFields.tsx (#28885)

grammarz 😉

AgnesToulet committed:
Dashboards / Folders: delete related data (permissions, stars, tags, versions, annotations) when deleting a dashboard or a folder (#28826)

* Dashboard: delete related data when deleting a dashboard or a folder

* fix migrations order

* apply PR feedback

AgnesToulet committed:
API: replace SendLoginLogCommand with LoginHook (#28777)

* API: replace SendLoginLogCommand with LoginHook

* LoginInfo: Query -> LoginUsername

aknuds1 committed:
Devenv: Fix Prometheus basic auth proxy (#28889)

* prometheus basic auth proxy: Pin base image version

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* prometheus basic auth proxy: Don't use host network mode

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* prometheus basic auth proxy: Enable logging

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* prometheus basic auth proxy: Fix Dockerfile location

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

Commit:
Adding terms to glossary (#28884)

* Adding terms to glossary

* requested edits and addition to style guide

* Update contribute/style-guides/documentation-style-guide.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Update docs/sources/getting-started/glossary.md

Co-authored-by: Marcus Olsson <marcus.olsson@hey.com>

* Update docs/sources/getting-started/glossary.md

Co-authored-by: Marcus Olsson <marcus.olsson@hey.com>

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>
Co-authored-by: Marcus Olsson <marcus.olsson@hey.com>

oddlittlebird committed:
Update documentation-style-guide.md (#28908)

ceh committed:
Chore: Rewrite test in GoConvey to stdlib and testify (#28918)

mckn committed:
Gauge: making sure threshold panel json is correct before render (#28898)

* making sure we work with a proper data structure.

* added test to verify functionality.

* removed unused variables.

Clarity-89 committed:
Grafana-UI: LoadingPlaceholder docs (#28874)

* Refactor props

* Add docs

papagian committed:
Chore: Fix spelling issue (#28904)

xlson committed:
Plugins: allow override when allowing unsigned plugins (#28901)

* Plugins: allow override when allowing unsigned plugins

* Update pkg/plugins/plugins.go

Co-authored-by: Emil Tullstedt <emil.tullstedt@grafana.com>

* Plugins: removed java-style setter

* Plugins: cleanup

* Update pkg/plugins/plugins.go

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

Co-authored-by: Emil Tullstedt <emil.tullstedt@grafana.com>
Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

zoltanbedi committed:
Update CHANGELOG.md (#28928)

This is not released yet.

ae3e committed:
CodeEditor: added support for javascript language (#28818)

mckn committed:
Variables: make sure that we support both old and new syntax for custom variables. (#28896)

dprokop committed:
Field color: handling color changes when switching panel types (#28875)

* FieldColor: Per panel settings to filter out supported modes

* Updates

* Updated solution

* Update panel plugin API for standard options support

* Update FieldColorConfigSettings interface

* Change color mode correctly when changing plugin type

* Render only applicable color modes in field color config editor

* Apply field config API changes

* TS fixes

Co-authored-by: Torkel Ödegaard <torkel@grafana.com>

dprokop committed:
Plugins signing: Fix docs urls (#28930)

aknuds1 committed:
setting: Fix tests on Mac (#28886)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

torkelo committed:
TestData: Fix issue with numeric inputs in TestData query editor (#28936)

aocenas committed:
Chore: Split Explore redux code into multiple sections (#28819)

* Split main reducer from item reducer

* Move query related redux to separate file

* Split more parts and tests

* Fix import

* Remove unused code

* Update public/app/features/explore/state/datasource.ts

Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

* Add section comments

* Rename ExploreItem to ExplorePane

* Fix imports

Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

zoltanbedi committed:
Chore: use jest without grunt (#28558)

* Chore: use jest without grunt

* Run no-focus-convey-tests and no-only-tests as well

* Update lib.star test-backend script with no-focus test to run first

* Lets see a failing test

* Revert "Lets see a failing test"

This reverts commit cb6c6ed67dd5403a16d4418aea3260ea8a286107.

* Update no-focus-convey test to a script

* Update scripts/lib.star

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update drone.yml

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

aocenas committed:
CloudWatch Logs: Change what we use to measure progress (#28912)

torkelo committed:
GraphNG: Using new VizLayout, moving Legend into GraphNG and some other refactorings (#28913)

* Graph refactorings

* Move legend to GraphNG and use new VizLayout

* Things are working

* remove unused things

* Update

* Fixed ng test dashboard

* Update

* More refactoring

* Removed plugin

* Upgrade uplot

* Auto size axis

* Axis scaling

* Fixed tests

* updated

* minor simplification

* Fixed selection color

* Fixed story

* Minor story fix

* Improve x-axis formatting

* Tweaks

* Update

* Updated

* Updates to handle timezone

* Updated

* Fixing types

* Update

* Fixed type

* Updated

wbrowne committed:
Short URL: Cleanup unvisited/stale short URLs (#28867)

* cleanup stale short urls

* refactor test case names

* service injection

* fix query

* add docs

* remove comma

oddlittlebird committed:
Docs: Fix links (#28945)

* Update getting-started.md

* Update configure-specific-fields.md

* Fixed links, added a test file to the shared folder

torkelo committed:
GraphNG: Restore focus option (#28946)

* GraphNG: Restore focus option

* Update snapshot

torkelo committed:
Grot: Added command/label to close feature requests with standard message (#28937)

* Grot: Added command/label to close feature requests with standard message

* add not implemented label

* Update and add metrics

* Updated wording

robbymilo committed:
Docs: updates for file-based menu (#28500)

* initial

* cleanup

* remove bad aliases

* cleanup, fix links

* add docs-file-based-command

* update docs

* update readme

* fix broken links

* fix spelling

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

torkelo committed:
GrafanaBot: Update labels and commands and adds docs  (#28950)

* GrafanaBot: Updates labels commands and adds docs

* Updated issue triage

* More updates

* fixed bullet list

torkelo committed:
Docs: Update bot docs

bergquist committed:
start tracking usage stats for tempo (#28948)

Signed-off-by: bergquist <carl.bergquist@gmail.com>

bergquist committed:
Tracing: log traceID in request logger (#28952)

Signed-off-by: bergquist <carl.bergquist@gmail.com>

ryantxu committed:
StatPanels: set default to last (#28617)

torkelo committed:
Docs: Update bot docs

torkelo committed:
Docs: Update PR workflow docs

torkelo committed:
Docs: Another minor edit

torkelo committed:
Docs: Workflow minor edit

torkelo committed:
Docs: Another workflow docs update

torkelo committed:
Docs: Minor docs update

robbymilo committed:
Docs: fix relref (#28977)

oddlittlebird committed:
Update getting-started-prometheus.md (#28502)

torkelo committed:
StatPanel: Fixed center of values in edge case scenarios (#28968)

aknuds1 committed:
Drone: Fix docs building (#28986)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aocenas committed:
LogsPanel: Don't show scroll bars when not needed (#28972)

daniellee committed:
docs: fix repeated dashboards link (#29002)

kaydelaney committed:
Backend: Adds route for well-known change password URL (#28788)

* Backend: Adds route for well-known change password URL

* Include 'dashboard/new' in backend routes

* Move index route handler registration out of "not logged in views" section

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

hugohaggmark committed:
Variables: Fixes loading with a custom all value in url (#28958)

hugohaggmark committed:
MSSQL: Support request cancellation properly (Uses new backendSrv.fetch Observable request API) (#28809)

* MSSQL: Support request cancellation properly (Uses new backendSrv.fetch Observable request API)

* Tests: fix test typings

aknuds1 committed:
Chore: Fix SQL related Go variable naming (#28887)

* Chore: Fix variable naming

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

torkelo committed:
StatPanels: Fixes auto min max when latest value is zero (#28982)

torkelo committed:
IntervalVariable: Fix variable tooltip (#28988)

systemcrash committed:
MySQL: Update README.md (#29003)

* Update README.md

sp

[skip ci] [ci skip]

* Update public/app/plugins/datasource/mysql/README.md

Co-authored-by: Torkel Ödegaard <torkel@grafana.com>

ivanahuckova committed:
Logs Panel: Fix inconsistent higlighting (#28971)

* Highlight filter expression in panel

* Remove coonsole.log

bergquist committed:
Tracing: Add setting for sampling server (#29011)

Signed-off-by: bergquist <carl.bergquist@gmail.com>
Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>
Co-authored-by: Emil Tullstedt <emil.tullstedt@grafana.com>

ivanahuckova committed:
Fix documentation (#28998)

aknuds1 committed:
Backend: Remove unused code (#28933)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

torkelo committed:
Thresholds: Fixes color assigned to null values (#29010)

* Thresholds: Fixes color assigned to null values

* Updates

davkal committed:
TraceViewer: Make sure it does not break when no trace is passed (#28909)

* TraceViewer: Make sure it does not break when no trace is passed

- exit trace conversion early when there is no trace data
- added test to cover missing trace for trace viewer

* Review feedback

* Use RTL for new test

* Exit early if trace or traceID is not there

torkelo committed:
Annotations: Fixes error when trying to create annotation when dashboard is unsaved (#29013)

sakjur committed:
Chore: Bump build pipeline version (#29023)

Commit:
Elasticsearch: interpolate variables in Filters Bucket Aggregation (#28969)

jackw committed:
Docs: Troubleshoot starting docker containers on Mac (#28754)

* docs(devenv): introduce troubleshooting section about mac osx and logs

* docs(devenv): tidy troubleshooting

* docs(devenv): move troubleshooting  headers up a level and expand on error messge

jackw committed:
Dashboard: remove usage of Legacyforms (#28707)

* refactor(dashboard): remove usage of legacyform components in sharemodal

* refactor(dashboard): replace legacyform components

* refactor(dashboard): remove ng-if and correct typo in content of sharesnapshot

* feat(grafana-ui): set displayName prop for Switch component

* refactor(dashboard): migrate TimePickerSettings legacyform components

* refactor(queryoptions): migrate switch and input to nextgen components

* refactor(sharesnapshot): prefer InlineFieldRow over gf-form-group

* refactor(shareembed): styling fixes

* refactor(timepickersettings): prefer double bang over nullish coalescing operator

* fix(grafana-ui): switch uses id prop if passed in

* feat: connect labels and switches with ids

ivanahuckova committed:
Docs: Add documentation about log levels (#28975)

* Add documentation about log level

* Update

* Update

* Update

* Add unknow to table

* Update docs/sources/explore/index.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

* Update docs/sources/panels/visualizations/logs-panel.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

avaly committed:
Explore: support ANSI colors in live logs (#28895)

Closes #28893

Co-authored-by: Ivana <ivana.huckova@gmail.com>

ivanahuckova committed:
Fix prometheus docs related to query variable (#29027)

torkelo committed:
MutableDataFrame: Remove unique field name constraint and values field index and unused/seldom used stuff  (#27573)

* MutableDataFrame needs work

* Updated users of values

* Update test

* Fixed ts issue

* Updated tests

xlson committed:
changelog 7.3.2 (#29038)

simianhacker committed:
Elasticsearch: Add Moving Function Pipeline Aggregation (#28131)

* Elasticsearch: Add Moving Function Pipeline Aggregation

* Removing support for moving average in 7.x

* Fixing getMetricAggTypes to handle undefined esVersion

* Adding moving_fn to go code

* Update public/app/plugins/datasource/elasticsearch/metric_agg.ts

Co-authored-by: Giordano Ricci <grdnricci@gmail.com>

* Adding test for esversion 70

* Removing default value for script, adding placeholder instead

* Fixing formatting

* Adding code for handling missing or obsolete aggregations

Co-authored-by: Giordano Ricci <grdnricci@gmail.com>

xlson committed:
latest 7.3.2 (#29041)

vishesh92 committed:
Fix links to logql docs (#29037)

torkelo committed:
GraphNG: Fixed axis measurements (#29036)

ceh committed:
Chore: Rewrite test helpers from GoConvey to stdlib (#28919)

papagian committed:
Fix for multi-value template variable for project selector (#29042)

domasx2 committed:
Logging: Log frontend errors (#28073)

* basic frontend  Sentry integration

* backend endpoint to capture sentry events

* WIP!

* log user email for frontend logs

* remove debug logging

* lint fixes

* Fix type exports & property names

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* additional struct naming fix

* rename log endpoint, config section & interface

* add sentry sample rate to config

* refac to use EchoSrv

* log user id

* backend tests

* tests for SentryEchoBackend

* sentry echo backend tests

* CustomEndpointTransport tests

* Update pkg/api/frontend_logging_test.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update conf/defaults.ini

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/api/frontend_logging_test.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* don't export unnecesasrily

* update go.sum

* get rid of Convey in tests, use stdlib

* add sentry config to sample.ini

* cleanup to set orig logging handler in test

* Apply suggestions from code review

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* PR feedback changes

* lock sentry version

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix linting issues (#28811)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

papagian committed:
AlertingNG: manage and evaluate alert definitions via the API (#28377)

* Alerting NG: prototype v2 (WIP)

* Separate eval package

* Modify eval alert definition endpoint

* Disable migration if ngalert is not enabled

* Remove premature test

* Fix lint issues

* Delete obsolete struct

* Apply suggestions from code review

* Update pkg/services/ngalert/ngalert.go

Co-authored-by: Kyle Brandt <kyle@grafana.com>

* Add API endpoint for listing alert definitions

* Introduce index for alert_definition table

* make ds object for expression to avoid panic

* wrap error

* Update pkg/services/ngalert/eval/eval.go

* Swith to backend.DataQuery

* Export TransformWrapper callback

* Fix lint issues

* Update pkg/services/ngalert/ngalert.go

Co-authored-by: Kyle Brandt <kyle@grafana.com>

* Validate alert definitions before storing them

* Introduce AlertQuery

* Add test

* Add QueryType in AlertQuery

* Accept only float64 (seconds) durations

* Apply suggestions from code review

* Get rid of bus

* Do not export symbols

* Fix failing test

* Fix failure due to service initialization order

Introduce MediumHigh service priority and assign it to backendplugin
service

* Fix test

* Apply suggestions from code review

* Fix renamed reference

Co-authored-by: Kyle Brandt <kyle@grafana.com>

marefr committed:
CloudWatch: Add support for AWS/ClientVPN metrics and dimensions (#29055)

Fixes #28778

tomdaly committed:
Add 'EventBusName' dimension to CloudWatch 'AWS/Events' namespace (#28402)

aknuds1 committed:
Plugins: Bring back coreplugin package (#29064)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

kaydelaney committed:
Test Datasource/Bug: Fixes division by zero in csv metric values scenario (#29029)

Closes #8705

kaydelaney committed:
Chore: Migrate Dashboard List panel to React (#28607)

* Chore: Migrate Dashlist to React
Closes #28491

torkelo committed:
GraphNG: Adding ticks test dashboard and improves tick spacing (#29044)

* GraphNG: Adding ticks test dashboard and improves tick spacing

* Added comments

* Trying to trigger CI

* Update

papagian committed:
Enhance automation text for missing information (#29052)

Ask to follow the issue template

oddlittlebird committed:
Update CODEOWNERS (#28906)

oddlittlebird committed:
Update README.md (#29075)

aknuds1 committed:
Chore: Convert API tests to standard Go lib (#29009)

* Chore: Convert tests to standard Go lib

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

Co-authored-by: Will Browne <wbrowne@users.noreply.github.com>

wbrowne committed:
Auth: Enable more complete credential chain for SigV4 default SDK auth option (#29065)

* Force more complete credential chain for default auth option

* simplify

* allow assume role for default

torkelo committed:
StatPanel: Fixes hanging issue when all values are zero (#29077)

* StatPanel: Fixes hanging issue when all values are zero

* Minor tweak

torkelo committed:
Graph: Fixes stacking issues like floating bars when data is not aligned  (#29051)

* Graph: Remove interpolation logic from graph stacking

* Improving tooltip

Clarity-89 committed:
Update react-testing-library (#29061)

* Update react-testing-library

* Update package.json

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

* Update lockfile

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

timpalpant committed:
Plugins: Implement testDatasource for Jaeger (#28916)

* Implement testDatasource for Jaeger

* Unit tests

* Run prettier

aknuds1 committed:
Data source proxy: Convert 401 from data source to 400 (#28962)

* Data source proxy: Convert 401 from data source to 400

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

ivanahuckova committed:
sort alphabetically unique labels, labels and parsed fields (#29030)

aocenas committed:
LogsPanel: Fix scrolling in dashboards (#28974)

hugohaggmark committed:
Chore: Bumps types for jest (#29098)

oddlittlebird committed:
Doc: Update documentation-style-guide.md (#29082)

* Update documentation-style-guide.md

* Update documentation-style-guide.md

torkelo committed:
Legend/GraphNG: Refactoring legend types and options (#29067)

* Legend/GraphNG: Refactoring legend types and options

* Rename label

* Minor update

* Fixed legend placement crash issue

* remove unused

* Minor tweaks and fixes

aknuds1 committed:
Drone: Verify Drone config at beginning of pipelines (#29071)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Plugin proxy: Handle URL parsing errors (#29093)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

ceh committed:
Chore: Require OrgId to be specified in delete playlist command (#29117)

aknuds1 committed:
Drone: Fix Drone config verification for enterprise on Windows (#29118)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

ryantxu committed:
DataFrames: add utility function to check if structure has changed (#29006)

* add common flag for knowing if the structure changes

* remove property

* fix test

* fix test

* update comment

* fix jsdoc comments

torkelo committed:
Automation: Adding version bump action

aknuds1 committed:
Chore: Upgrade Go dev tools (#29124)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

ceh committed:
Chore: Rewrite preferences test from GoConvey to stdlib and testify (#29129)

torkelo committed:
Automation: Adds new changelog actions (#29142)

* Automation: Adds new changelog actions

* Update

* Add newline

* added link

* Another fix

alexanderzobnin committed:
Provisioning: always pin app to the sidebar when enabled (#29084)

amattheisen committed:
build paths in an os independent way (#29143)

Use filepath instead of path to manipulate filename paths in a way compatible with the target operating system-defined file paths.

domasx2 committed:
respect fronted-logging.enabled flag (#29107)

papagian committed:
Fix authomation text: remove hyphen (#29149)

torkelo committed:
Snapshots: Fixes so that dashboard snapshots show data when using Stat, Gauge, BarGauge or Table panels (#29031)

cinaglia committed:
Fix panic when using complex dynamic URLs in app plugin routes  (#27977)

* remove unused function to interpolate URLs

* share function to add headers between ds/plugin proxies

* stop performing unnecessary plugin setting lookup

* fix bug causing runtime errors when using complex templated URLs

* lower case util functions not used outside of pluginproxy package

* change test URL to a (valid) dummy URL to make intent clearer

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Chore: Fix linting issues caught by ruleguard (#28799)

* Chore: Fix linting issues caught by ruleguard

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Improve error check

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
API: Rewrite tests from goconvey (#29091)

* API: Rewrite tests from goconvey

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Chore: Remove unused Go code (#28852)

* Chore: Remove more unused Go code

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Chore: Upgrade Go etc in build images (#29157)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 7.3.3 (#29159)

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 7.4.0 (#29160)

aknuds1 committed:
Drone: Upgrade build pipeline tool and build image (#29161)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

torkelo committed:
Docs: Fixes what'new menu and creates index page, adds first draft of release notes to docs (#29158)

* Docs: Fixes what'new menu and creates index page, adds first draft of release notes to docs

* Minor tweaks

* Fixed missing letter

* Updates

* Updated

Co-authored-by: Drone <drone@grafana.com>

aknuds1 committed:
Backend: Rename variables for style conformance (#29097)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

Commit:
changed link format from MD to HTML (#29163)

azhiltsov committed:
SQL: Define primary key for tables without it (#22255)

The annotation_tag and alert_rule_tag tables did not have
PRIMARY KEY defined what cause problems with migration to
MariaDB with Galera setup with
 innodb_force_primary_key=1
Or MySQL > 8.0.13 with
 sql_require_primary_key=ON

Which can manifest as follows:

MariaDB
Error 1173: This table type requires a primary key

MySQL
ERROR 3750 (HY000): Unable to create or change a table
without a primary key, when the system variable 'sql_require_primary_key' is set.

Extra reading for curious:
https://jfg-mysql.blogspot.com/2017/08/danger-no-pk-with-RBR-and-mariadb-protection.html

Fixes #12971

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

torkelo committed:
Release: Adding release notes for 7.3.3 (#29168)

* Release: Adding release notes for 7.3.3

* Updates

karlatkinson committed:
CloudWatch: added HTTP API Gateway specific metrics and dimensions (#28780)

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 7.3.3 (#29169)

mckn committed:
DashboardLinks: will only refresh dashboard search when changing tags for link. (#29040)

* fixing so we dont run multiple dashboard links searches when changing variables.

* changed so we fetch the list when open the dashboard links dropdown.

* removed verification of unneccesary requests.

aknuds1 committed:
Chore: Upgrade grafana/build-ci-deploy image to latest Go (#29171)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

dprokop committed:
Table: Fix incorrect condtition for rendering table filter (#29165)

aknuds1 committed:
Drone: Fix publish-packages invocation (#29179)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
playlist: Improve test (#29120)

* playlist: Improve test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

bergquist committed:
trace user login and datasource name instead of id (#29183)

id's are not very helpful when debugging a system.

Signed-off-by: bergquist <carl.bergquist@gmail.com>

krvajal committed:
Panels: fix positioning of the header title (#29167)

the menu toggle icon is now absolute positioned and it no longer  affects the centering of the menu title

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 7.3.3 (#29189)

dprokop committed:
Refactor declarative series configuration to a config builder (#29106)

* Wip: refactor declarative series configuration to a config builder

* Fix plugins initialization

* Config builder reorg and tests

* Typecheck

* Update packages/grafana-ui/src/components/uPlot/context.ts

* Scales config tweak

* Temp disable tests

* Disable some tests temporarily

torkelo committed:
Automation: Adds GitHub release action (#29194)

* Automation: Adds GitHub release action

* Updated wording

* minor update

dprokop committed:
Introduce eslint-plugin-react (#29053)

* Eslint: allign with latest grafana-eslint-config

* fix ts

* Bump @grafana-eslint-congig version

hugohaggmark committed:
Variables: Adds new Api that allows proper QueryEditors for Query variables (#28217)

* Initial

* WIP

* wip

* Refactor: fixing types

* Refactor: Fixed more typings

* Feature: Moves TestData to new API

* Feature: Moves CloudMonitoringDatasource to new API

* Feature: Moves PrometheusDatasource to new Variables API

* Refactor: Clean up comments

* Refactor: changes to QueryEditorProps instead

* Refactor: cleans up testdata, prometheus and cloud monitoring variable support

* Refactor: adds variableQueryRunner

* Refactor: adds props to VariableQueryEditor

* Refactor: reverted Loki editor

* Refactor: refactor queryrunner into smaller pieces

* Refactor: adds upgrade query thunk

* Tests: Updates old tests

* Docs: fixes build errors for exported api

* Tests: adds guard tests

* Tests: adds QueryRunner tests

* Tests: fixes broken tests

* Tests: adds variableQueryObserver tests

* Test: adds tests for operator functions

* Test: adds VariableQueryRunner tests

* Refactor: renames dataSource

* Refactor: adds definition for standard variable support

* Refactor: adds cancellation to OptionPicker

* Refactor: changes according to Dominiks suggestion

* Refactor:tt

* Refactor: adds tests for factories

* Refactor: restructuring a bit

* Refactor: renames variableQueryRunner.ts

* Refactor: adds quick exit when runRequest returns errors

* Refactor: using TextArea from grafana/ui

* Refactor: changed from interfaces to classes instead

* Tests: fixes broken test

* Docs: fixes doc issue count

* Docs: fixes doc issue count

* Refactor: Adds check for self referencing queries

* Tests: fixed unused variable

* Refactor: Changes comments

hugohaggmark committed:
Tests: Adds expects for observables (#28929)

* Tests: Adds expects for observables

* Refactor: renames matcher

* Chore: changed jest version and removed as dev dependency

* Refactor: Changes after PR review

* wip

* Chore: revert

* Chore: Adds jest-matcher-utils

* Chore: fixed merge error

* Tests: Removed inline snapshots as they are env dependent

* Tests: missed a snapshot

* Refactor: Removed type file after PR comments

dprokop committed:
ESlint: React fixes part 1 (#29062)

* Eslint: allign with latest grafana-eslint-config

* fix ts

* Fix react/jsx-key

* Fix react/no-children-prop

* Fix react/jsx-no-target-blank

wbrowne committed:
Permissions: Validate against Team/User permission role update (#29101)

* validate against role field update

* lowercase error string

* make all msgs consistent style

* fix wording

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* sayonara simple json

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Backend: Fix build (#29206)

* Backend: Fix build

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>
